test/e2e/elasticache/service_bootstrap.py

```diff
@@ -0,0 +1,147 @@
+
+    return sg_response['GroupId']
+
+def create_user_group() -> str:
+    ec = boto3.client("elasticache")
+
+    usergroup_id = random_suffix_name("ack-ec-usergroup", 32)
+    _ = ec.create_user_group(UserGroupId=usergroup_id,
+                                    Engine="Redis",
+                                    UserIds=["default"])
+    logging.info(f"Creating ElastiCache User Group {usergroup_id}")
+    assert wait_usergroup_active(usergroup_id)
+
+    return usergroup_id
+
+def create_kms_key() -> str:
+    kms = boto3.client("kms")
+
+    response = kms.create_key(Description="Key for ACK ElastiCache tests")
+    key_id = response['KeyMetadata']['KeyId']
+    logging.info(f"Created KMS key {key_id}")
+
+    return key_id
+
+# create global replication group
+def create_grg() -> str:
+
+    # primary should be in a different region than default
+    grg_primary_region = "us-west-2"
+    default_region = get_aws_region()
+    if default_region == "us-west-2":
+        grg_primary_region = "us-east-1"
+
+    # create primary cluster for GRG
+    ec = boto3.client("elasticache", region_name=grg_primary_region)
+    grg_primary_id = random_suffix_name("ack-grg-primary-cluster", 32)
+    _ = ec.create_replication_group(
+        ReplicationGroupId=grg_primary_id,
+        ReplicationGroupDescription="Primary cluster gor GRG",
+        Engine="Redis",
+        CacheNodeType="cache.r6g.large"
+    )
+
+    # create GRG using newly created primary
+    grg_suffix_id = random_suffix_name("ack-test-grg", 32)
+    response = ec.create_global_replication_group(
+        GlobalReplicationGroupIdSuffix=grg_suffix_id,
+        PrimaryReplicationGroupId=grg_primary_id
+    )
+
+    # generated ID differs from specified "suffix ID" so retrieve it from the response
+    grg_id = response['GlobalReplicationGroup']['GlobalReplicationGroupId']
+    logging.info(f"Created Global Replication Group {grg_id}")
+
+    return grg_id
+
+# create a cache cluster, snapshot it, and return the snapshot name
+def create_cc_snapshot():
+    ec = boto3.client("elasticache")
+
+    cc_id = random_suffix_name("ack-cache-cluster", 32)
+    _ = ec.create_cache_cluster(
+        CacheClusterId=cc_id,
+        NumCacheNodes=1,
+        CacheNodeType="cache.m6g.large",
+        Engine="redis"
+    )
+    waiter = ec.get_waiter('cache_cluster_available')
+    waiter.wait(CacheClusterId=cc_id)
+    logging.info(f"Created cache cluster {cc_id} for snapshotting")
+
+    snapshot_name = random_suffix_name("ack-cc-snapshot", 32)
+    _ = ec.create_snapshot(
+        CacheClusterId=cc_id,
+        SnapshotName=snapshot_name
+    )
+    assert wait_snapshot_available(snapshot_name)
+
+    return snapshot_name
+
+@dataclass
+class BootstrapResources:
+    SnsTopicARN: str
+    SecurityGroupID: str
+    UserGroupID: str
+    KmsKeyID: str
+    GrgID: str
+    SnapshotName: str
+
+def service_bootstrap() -> dict:
+    logging.getLogger().setLevel(logging.INFO)
```

Comment:
I was initially following what the Sagemaker team has done. After looking into it more, they have some logging.debug statements that this statement will actually filter out (e.g. logging the dictionary representation of the injected resource yaml) whereas I currently only have logging.info statements. I'll either do the same or just remove the statements altogether.

Reply:
@RedbackThomson see above...

Reply:
We have debugging statements because as part of running the tests you can override the logger level - as provided by the PyTest library I think?

https://github.com/aws-controllers-k8s/community/blob/main/test/e2e/run-tests.sh#L57

So if you want to add debugging outputs for your own sanity when testing the new tests, you should be able to adjust it by setting `PYTEST_LOG_LEVEL=debug`